### PR TITLE
fix(youtubevanced): hide `ads shelf` and `premium promotional banner`

### PR DESCRIPTION
**Pull request by johnconner122**

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/HideAdsPatch.kt [hunk 1]
```diff
@@ -46,7 +46,9 @@ class HideAdsPatch : BytecodePatch(
                     "video_display_full_layout",
                     "hero_promo_image",
                     "statement_banner",
-                    "primetime_promo"
+                    "primetime_promo",
```
[the remark this answers is not in the document]
Comment (johnconner122) [reply src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/HideAdsPatch.kt:49]: Not at the moment, but it sometimes appears when I am connected to VPN.

**Comment by oSumAtrIX** [comment]
In that case please also open a PR for YouTube

**Comment by oSumAtrIX** [comment]
Is this up to date?

File: src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/HideAdsPatch.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/HideAdsPatch.kt:49]: In that case please also open a PR for YouTube

File: src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/HideAdsPatch.kt [hunk 2]
```diff
@@ -46,7 +46,9 @@ class HideAdsPatch : BytecodePatch(
...
+                    "carousel_footered_layout",
+                    "feature_grid_interstitial"
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/HideAdsPatch.kt:51]: What do these block?

Comment (johnconner122) [reply src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/HideAdsPatch.kt:51]: YouTube sometimes offers premium trial, it blocks that banner.

Comment (oSumAtrIX) [reply src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/HideAdsPatch.kt:51]: Two strings were added, what about the other?

Comment (johnconner122) [reply src/main/kotlin/app/revanced/patches/youtubevanced/ad/general/patch/HideAdsPatch.kt:51]: - `primetime_promo` blocks YouTube TV ads


image

**Review by oSumAtrIX — APPROVED** [review]

**Pull request merged**